src/redux/todoSlice.jsx

```diff
@@ -0,0 +1,60 @@
+import { createSlice, nanoid } from "@reduxjs/toolkit";
+const initialState = {
+  todos: JSON.parse(localStorage.getItem("todos")) || [],
+  filter: localStorage.getItem("filter") || "all",
+};
+
+export const todoSlice = createSlice({
+  name: "todo",
+  initialState: initialState,
+  reducers: {
+    addTodo: (state, action) => {
+      const newTodo = {
+        id: nanoid(),
+        text: action.payload,
+        completed: false,
+      };
+      state.todos.push(newTodo);
+      localStorage.setItem("todos", JSON.stringify(state.todos));
```

Comment:
set the local storage data when there is some change in todo array
you can use useEffect instead of writing it again and again